### PR TITLE
80 return generic info when failing to read calculation

### PR DESCRIPTION
**Pull request by YHordijk**
The `tcutility.results.read` function should now no longer crash. It will also use slurm now to detect the job status (e.g. pending, running, completing, etc.).

### CODE REVIEW

File: pyproject.toml
Comment (SiebeLeDe) [inline pyproject.toml:1]: Aha did not know this worked. I tried it before but then it crashed. Nice

File: src/tcutility/results/cache.py
Comment (SiebeLeDe) [inline src/tcutility/results/cache.py:1]: I can predict some annoying issues in the future when we get "None" from failed jobs (calculation folders). Is that correct?

File: src/tcutility/slurm.py
Comment (SiebeLeDe) [inline src/tcutility/slurm.py:1]: Could you provide more information in the docstrings? For example, the "workdir_info" is vague to me now. 
Also, I wonder if we need to implement commands like squeue that can easily be achieved with shell commands on the cluster.

File: src/tcutility/results/cache.py
Comment (YHordijk) [reply src/tcutility/results/cache.py:1]: Indeed, cache should return None for jobs that failed. Alternatively we can raise an error. What do you think is the best approach here?

File: src/tcutility/slurm.py
Comment (YHordijk) [reply src/tcutility/slurm.py:1]: I have added some extra docstrings for the current functions!

**Review by SiebeLeDe — APPROVED** [review]

**Pull request merged**